cabal-install/Distribution/Client/ProjectConfig.hs

```diff
@@ -418,38 +418,19 @@ renderBadProjectRoot (BadProjectRootExplicitFile projectFile) =
 readProjectConfig :: Verbosity -> Flag FilePath -> DistDirLayout -> Rebuild ProjectConfig
 readProjectConfig verbosity configFileFlag distDirLayout = do
     global <- readGlobalConfig             verbosity configFileFlag
-    local  <- readProjectLocalConfig       verbosity distDirLayout
+    hasCabalProjectFile <- liftIO $ doesFileExist projectFile
```

Comment:
I liked it better as it was before with 

```
thing1 <- read this
thing2 <- read that
thing3 <- read that 
... 
```

The logic here looks a bit out of place.

Reply:
you could lift `addProjectFileProvenance` to top level for example to reduce the duplication that way.

Reply:
The downside to that is that the reader might very well expect the same kind of logic behind the three `read` functions, especially when they have so similar names (`readProjectLocalConfig`, `readProjectLocalFreezeConfig`, `readProjectLocalExtraConfig`).

The `addProjectFileProvenance` change is not necessary afaict.

Reply:
Right, would it make sense to pass the implicit config as a parameter in case the file doesn't exist? Then you could handle the "file does not exist case"  in `readProjectFile`. Does that make sense?